### PR TITLE
Add algotirhm LCA, manacher, inverse pair count calculator

### PR DESCRIPTION
**Pull request by smallbearrr**

### CODE REVIEW

**Comment by smallbearrr** [comment]
it checked failed, what should i change that can let it pass the check? @Lampese

**Comment by Lampese** [comment]
> it checked failed, what should i change that can let it pass the check? @Lampese 

I will start to repair the Action, which will not affect the review of pr.

File: inversePair/inversePair_test.mbt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+test {
+  let arr = [1, 2, 3, 4, 1]
+  inspect!(@inversePair.inverse_pair(arr), content="3")
+  let arr2 = [5, 4, 3, 2, 1]
+  inspect!(@inversePair.inverse_pair(arr2), content="10")
+}
```
Comment (Lampese) [inline inversePair/inversePair_test.mbt:6]: Did you use `moon fmt`?

File: inversePair/inversePair.mbt [hunk 2]
```diff
@@ -0,0 +1,63 @@
+fn merge[T : Compare](a : Array[T], b : Array[T], l : Int, r : Int) -> Int {
```
Comment (Lampese) [inline inversePair/inversePair.mbt:1]: Filename should use snake format like `inverse_pair`.

**Review by Lampese — CHANGES_REQUESTED** [review]

**Pull request merged**